src/2d/shallow/landspill/point_source/point_source_module.f90

```diff
@@ -0,0 +1,296 @@
+
+        ! code
+        call destructor(this)
+
+        read(funit, *) this%id
+        read(funit, *) this%coord
+        read(funit, *) this%nt
+
+        allocate(this%t(this%nt), this%v_rate(this%nt))
+
+        read(funit, *) this%t
+        read(funit, *) this%v_rate
+
+        ! TODO: make sure the t array is sorted and v_rate is sorted based on t
+
+    end subroutine init_from_funit
+
+    !> @brief Print information of an instance of PointSource.
+    !!
+    !! Print information of an instance of PointSource to specified file unit.
+    !!
+    !! @param[in] this an instance of PointSource
+    !! @param[in] iounit see intrinsic write function
+    !! @param[in] iotype see intrinsic write function
+    !! @param[in] v_list see intrinsic write function
+    !! @param[out] stat see intrinsic write function
+    !! @param[in, out] msg see intrinsic write function
+    subroutine writef(this, iounit, iotype, v_list, stat, msg)
+        ! variable declaration
+        class(PointSource), intent(in):: this
+        integer(kind=4), intent(in):: iounit
+        character(*), intent(in)::iotype
+        integer(kind=4), intent(in):: v_list(:)
+        integer(kind=4), intent(out):: stat
+        character(*), intent(inout):: msg
+        character:: n, t
+
+        n = new_line(t) ! n is the "new line" character
+        t = achar(9) ! t is the character for a tab
+
+        ! code
+        write(iounit, *, iostat=stat, iomsg=msg) &
+            n, this%id, t, t, t, t, "=: id # ID of this point source"
+        write(iounit, *, iostat=stat, iomsg=msg) &
+            n, this%coord, "=: coord # coordinates"
+        write(iounit, *, iostat=stat, iomsg=msg) &
+            n, this%nt, t, t, t, t, "=: n_times # number of time segments"
+        write(iounit, *, iostat=stat, iomsg=msg) &
+            n, this%t, "=: end_times # end times of segments"
+        write(iounit, *, iostat=stat, iomsg=msg) &
+            n, this%v_rate, "=: vol_rates # volumetric rates of segments"
+
+    end subroutine writef
+
+    !> @brief Read formatted data block and re-initialize the instance.
+    !!
+    !! Read formatted data block and re-initialize the instance.
+    !!
+    !! @param[in] this an instance of PointSource
+    !! @param[in] iounit see intrinsic write function
+    !! @param[in] iotype see intrinsic write function
+    !! @param[in] v_list see intrinsic write function
+    !! @param[out] stat see intrinsic write function
+    !! @param[in, out] msg see intrinsic write function
+    subroutine readf(this, iounit, iotype, v_list, stat, msg)
+        ! variable declaration
+        class(PointSource), intent(inout):: this
+        integer(kind=4), intent(in):: iounit
+        character(*), intent(in)::iotype
+        integer(kind=4), intent(in):: v_list(:)
+        integer(kind=4), intent(out):: stat
+        character(*), intent(inout):: msg
+
+        ! code
+        write(*, *) "Direct read of this object is prohibited!"
```

Comment:
I tend to prefer to explicitly use `print *,` if this is really intended to go to the console.

Reply:
Yes. I think using `print *,` indeed provides a more clear hint that this is going to the console. I'll work on this one.

Reply:
Done in 61455d9.